Lib/functools.py

```diff
@@ -942,13 +945,27 @@ def register(self, cls, method=None):
         return self.dispatcher.register(cls, func=method)
 
     def __get__(self, obj, cls=None):
+        if self._all_weakrefable_instances:
+            try:
+                _method = self._method_cache[obj]
+            except TypeError:
+                self._all_weakrefable_instances = False
```

Comment:
Hmm, I wonder if this would have been a good idea, to keep memory usage down for the slow path (since in the slow path, we don't really need the `WeakKeyDictionary` at all)

```suggestion
                self._all_weakrefable_instances = False
                del self._method_cache
```

@corona10, what do you think? Does it matter?

Reply:
One could even delay the creation of the Weakkeydictionary untill the first invocation of `__get__`. The code will look a bit odd though, a simple `del` looks cleaner.

Reply:
Yeah, I think I prefer the approach with `del`